atom/browser/api/atom_api_desktop_capturer.cc

```diff
@@ -169,8 +172,14 @@ void DesktopCapturer::UpdateSourcesList(DesktopMediaList* list) {
       std::vector<std::string> device_names;
       // Crucially, this list of device names will be in the same order as
       // |media_list_sources|.
-      webrtc::DxgiDuplicatorController::Instance()->GetDeviceNames(
-          &device_names);
+      bool success =
+          webrtc::DxgiDuplicatorController::Instance()->GetDeviceNames(
+              &device_names);
+
+      if (!success)
+        args->ThrowError(
```

Comment:
Should this actually be an error or can we somehow fail softly and indicate that the `display_id` property won't be available but still return the other information?

Reply:
i don't think so, because all subsequent info surfaced on windows hinges on having those names